### DISCUSSION TITLE
Improve error message for "MethodError: factornode"

### DISCUSSION THREAD

**Discussion by fonsp** [ReactiveBayes/RxInfer.jl#376] [Q&A]
I created a model with a distribution that is not supported (Cauchy):

```julia
@model function lighthouse_model(observations, alpha_prior, beta_prior)
	alpha ~ alpha_prior
	beta ~ beta_prior

	for i in eachindex(observations)
		observations[i] ~ Cauchy(alpha, beta)
	end
end
```

The error message from `infer` is below, with a very long stack trace:
```
MethodError: no method matching factornode(::ReactiveMP.UndefinedNodeFunctionalForm, ::Type{Distributions.Cauchy}, ::Vector{Tuple{Symbol, ReactiveMP.AbstractVariable}}, ::Tuple{Vector{Int64}, Vector{Int64}})

The function `factornode` exists, but no method is defined for this combination of argument types.

Closest candidates are:
  factornode(::ReactiveMP.UndefinedNodeFunctionalForm, ::F, ::Any, ::Any) where F<:Function
   @ ReactiveMP ~/.julia/packages/ReactiveMP/1xtcb/src/nodes/predefined/delta/delta.jl:112
  factornode(::ReactiveMP.PredefinedNodeFunctionalForm, ::F, ::I, ::Any) where {F, I}
   @ ReactiveMP ~/.julia/packages/ReactiveMP/1xtcb/src/nodes/nodes.jl:190
  factornode(::F, ::I, ::Any) where {F, I}
   @ ReactiveMP ~/.julia/packages/ReactiveMP/1xtcb/src/nodes/nodes.jl:184
  ...
```

# Improved error message
The error message can be improved! The helpful information (that I got from Dmitry) was: "The distribution Cauchy is not supported by RxInfer" and it would be nice to have this in the error message. The current error message is not helpful because I do not know why `factornode` was called (or what it is).

I found that `Base.Experimental.register_error_hint` is useful for improving MethodError messages, see eg https://github.com/JuliaPlots/Plots.jl/pull/4733

# Example notebook
[no method factornode.html.zip](https://github.com/user-attachments/files/17661049/no.method.factornode.html.zip)

**Comment by bvdmitri** [comment]
Hey @fonsp thanks for the feedback! I made a PR https://github.com/ReactiveBayes/ReactiveMP.jl/pull/431 for this
